x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/DomainSplitFunction.java

```diff
@@ -5,279 +5,285 @@
  */
 package org.elasticsearch.xpack.ml.utils;
 
+import org.apache.logging.log4j.LogManager;
 import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.logging.DeprecationLogger;
 
 import java.io.InputStream;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
+import java.util.StringJoiner;
 
 public final class DomainSplitFunction {
 
-    public static final String function;
-    public static final Map<String, Object> params;
+    private static final DeprecationLogger DEPRECATION_LOGGER =
```

Comment:
Now this file split function is java it should have a unit test. @rjernst I may be able to find the legacy test from the prelert repo are you ok with me adding that to this change (if I can find it)

Reply:
That would be great!